common/src/main/java/io/pravega/common/io/serialization/RevisionDataOutput.java

```diff
@@ -23,26 +24,36 @@
  * An instance of a RevisionDataOutput is created for each Serialization Revision and closed when that Revision's serialization
  * is done - it is not shared between multiple revisions.
  *
- * This interface is designed to serialize data that can be consumed using RevisionDataInput.
+ * This interface is designed to serialize data that can be consumed using {@link RevisionDataInput}.
  */
 public interface RevisionDataOutput extends DataOutput {
     /**
-     * Maximum value that can be encoded using writeCompactLong().
+     * Maximum value that can be encoded using {@link #writeCompactLong).
      */
     long COMPACT_LONG_MAX = 0x3FFF_FFFF_FFFF_FFFFL;
 
     /**
-     * Minimum value that can be encoded using writeCompactLong().
+     * Minimum value that can be encoded using {@link #writeCompactLong}.
      */
     long COMPACT_LONG_MIN = 0L;
 
     /**
-     * Maximum value that can be encoded using writeCompactInt().
+     * Maximum value that can be encoded using {@link #writeCompactSignedLong}.
```

Comment:
If the acceptable range is  `[-2^61, 2^61)`,  then shouldn't the maximum value that can be encoded be `0x1FFF_FFFF_FFFF_FFFFL - 1`?

It is just a matter of convention, but my natural reaction is to think that the maximum is inclusive as opposed to the maximum exclusive of the range. The options I suggest are:

1- Make the maximum `0x1FFF_FFFF_FFFF_FFFFL - 1`
2- Clarify in the comment that this is the top of the range, not inclusive (the range is open-ended).

Reply:
Good points. I adjusted the values (subtracted 1) and fixed the code to handle it.